File: 3264. Final Array State After K Multiplication Operations I.py
class Solution:
    def getFinalState(self, nums: List[int], k: int, multiplier: int) -> List[int]:
        minHeap = [(val, idx) for idx, val in enumerate(nums)]
        heapify(minHeap)

        for _ in range(k):
            val, idx = heappop(minHeap)
            heappush(minHeap, (val * multiplier, idx))
        
        ans = [0] * len(nums)
        while minHeap:
            val, idx = heappop(minHeap)
            ans[idx] = val
        return ans

File: 3264. Final Array State After K Multiplication Operations I.cpp
typedef pair<long, int> pi;
class Solution {
public:
    vector<int> getFinalState(vector<int>& nums, int k, int multiplier) 
    {
        priority_queue<pi, vector<pi>, greater<pi>>minHeap;
        
        int n = nums.size();
        for(int i=0;i<n;i++)
            minHeap.push({nums[i], i});
        
        while(k--)
        {
            auto [val, idx] = minHeap.top(); minHeap.pop();
            minHeap.push({val * multiplier, idx});
        }

        vector<int>ans(n);
        while(!minHeap.empty())
        {
            auto [val, idx] = minHeap.top(); minHeap.pop();
            ans[idx] = val;
        }
        return ans;
    }
};